python/src/array.cpp

```diff
@@ -610,6 +656,38 @@ void init_array(py::module_& m) {
           )pbdoc")
       .def("__getitem__", mlx_get_item)
       .def("__setitem__", mlx_set_item)
+      .def_property_readonly(
+          "at",
+          [](const array& a) { return ArrayAt(a); },
+          R"pbdoc(
+            Returns a helper object that allows to apply updates at specific indices.
```

Comment:
I wonder if explaining this as a proxy object will just confuse people?

Might be worth simplifying to:

`Used to apply updates at the given indices`

Just a thought...not sure

Reply:
Done.